web/src/context/installerL10n.tsx

```diff
@@ -210,71 +187,61 @@ function reload(newLanguage) {
  * The format of the language tag in the query parameter follows the
  * [RFC 5646](https://datatracker.ietf.org/doc/html/rfc5646) specification.
  *
- * @param {object} props
- * @param {React.ReactNode} [props.children] - Content to display within the wrapper.
+ * @param props
+ * @param [props.children] - Content to display within the wrapper.
  *
  * @see useInstallerL10n
  */
-function InstallerL10nProvider({ children }) {
+function InstallerL10nProvider({ children }: { children?: React.ReactNode }) {
   const { connected } = useInstallerClientStatus();
   const [language, setLanguage] = useState(undefined);
   const [keymap, setKeymap] = useState(undefined);
-  const [backendPending, setBackendPending] = useState(false);
   const { cancellablePromise } = useCancellablePromise();
 
-  const storeInstallerLanguage = useCallback(
-    async (newLanguage) => {
-      if (!connected) {
-        setBackendPending(true);
-        return false;
-      }
-
-      const config = await cancellablePromise(fetchConfig());
-      const currentLanguage = languageFromLocale(config.uiLocale);
-
-      if (currentLanguage !== newLanguage) {
-        // FIXME: fallback to en-US if the language is not supported.
-        await cancellablePromise(updateConfig({ uiLocale: languageToLocale(newLanguage) }));
-        return true;
-      }
+  const syncBackendLanguage = useCallback(async () => {
+    const config = await cancellablePromise(fetchConfig());
```

Comment:
Question, do we still needing cancellablePromise?

Reply:
Good point. No, I think we do not need it.